cdap-api/src/main/java/co/cask/cdap/api/schedule/Schedules.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.schedule;
+
+import co.cask.cdap.internal.schedule.StreamSizeSchedule;
+import co.cask.cdap.internal.schedule.TimeSchedule;
+
+/**
+ * Utility class to create {@link Schedule} objects.
+ */
+public class Schedules {
+  
+  /**
+   * Build a time-based schedule.
+   *
+   * @param name name of the schedule
+   * @param description description of the schedule
+   * @param cronExpression cron expression for the schedule
+   * @return a schedule based on the given {@code cronExpression}
+   */
+  public static Schedule createTimeSchedule(String name, String description, String cronExpression) {
+    return new TimeSchedule(name, description, cronExpression);
+  }
+
+  /**
+   * Build a schedule based on data availability.
+   *
+   * @param name name of the schedule
+   * @param description description of the schedule
+   * @param source source of data the schedule is based on
+   * @param sourceName name of the source of data the schedule is based on
+   * @param dataTriggerMB the size of data, in MB, that the source has to receive to trigger an execution. It can be
+   *                      null if this parameter is irrelevent for the {@code dataType} given.
```

Comment:
I don't like API that needs user to pass in `null` in certain situation. If all we supported right now is size, make that an `int`. We can revise when we add support for other types of data schedule.